scripts/source_stats.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+"""Script to compute statistics about source code."""
+
+from __future__ import annotations
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import argparse
+from alive_progress import alive_bar  # type:ignore
+from multiprocessing import Pool
+import re
+import termplotlib as tpl  # type:ignore
+from pathlib import Path
+from typing import Dict, List, Optional
+from dataclasses import dataclass, field, asdict
+
+BLANK_RE = re.compile(r"\s*")
+COMMENT_RE = re.compile(r"\s*///*\s*")
+LINE_RE = re.compile(
+    r"""
+    (?P<class_intro>\b(class|struct)\s+(?P<class_name>\w+)\b)|
+    (?P<end_open_curly>{\s*(?P<open_curly_trailing_comment>//.*)?)|
+    (?P<trailing_comment>//.*)|
+    (?P<id>\b\w+\b)
+""",
+    re.X,
+)
+
+
+@dataclass
+class _Stats:
```

Comment:
Using `_` prefix seems like extra noise for a script like this.

Reply:
Done.